modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/PicklistObjectFieldBusinessType.java

```diff
@@ -177,9 +170,23 @@ public void predefineObjectFieldSettings(
 			ObjectStateFlow objectStateFlow =
 				objectFieldSetting.getObjectStateFlow();
 
-			_objectStateFlowLocalService.addObjectStateFlow(
-				newObjectField.getUserId(), newObjectField.getObjectFieldId(),
-				objectStateFlow.getObjectStates());
+			if (oldObjectField != null) {
+				ObjectStateFlow serviceBuilderObjectStateFlow =
+					_objectStateFlowLocalService.
+						fetchObjectFieldObjectStateFlow(
+							oldObjectField.getObjectFieldId());
+
+				_objectStateFlowLocalService.updateObjectStateFlowTransitions(
+					newObjectField.getUserId(),
+					serviceBuilderObjectStateFlow.getObjectStateFlowId(),
+					objectStateFlow.getObjectStates());
+			}
+			else {
+				_objectStateFlowLocalService.addObjectStateFlow(
+					newObjectField.getUserId(),
+					newObjectField.getObjectFieldId(),
+					objectStateFlow.getObjectStates());
+			}
```

Comment:
@G-Sa could you explain to me this new behavior?

Reply:
The idea is to keep adding the ObjectStateFlow as usual if we are inside a "post" flow, but if we are in an "put" flow we should also be able to update the transitions of the flow, not only the possible states that that flow has.

The `addObjectStateFlow` creates states and transitions already, so no need to touch the transitions there, but before that we would only create/update the states during a "put" flow, but the transitions would be left as they were, which could lead to situations where new states introduced in a put would transition to nowhere